unittests/savanna_disaster_recovery.cpp

```diff
@@ -0,0 +1,270 @@
+#include "savanna_cluster.hpp"
+
+using namespace eosio::chain;
+using namespace eosio::testing;
+
+BOOST_AUTO_TEST_SUITE(savanna_disaster_recovery)
+
+// ---------------------------------------------------------------------------------------------------
+BOOST_FIXTURE_TEST_CASE(node_goes_down, savanna_cluster::cluster_t) try {
+   auto& A=_nodes[0]; auto& C=_nodes[2];
+
+   C.close();
+   BOOST_REQUIRE_EQUAL(A.lib_advances_by([&]() { A.produce_blocks(4);  }), 4); // lib still advances with 3 finalizers
```

Comment:
Can we verify only 3 finalizers (not including C) are voting?

Reply:
We don't have an API for doing this directly... let me see if I can add one.